tools/Deployment/lib/common.ts

```diff
@@ -0,0 +1,157 @@
+            }
+        });
+
+        let buffer = zip.generate({ type: "nodebuffer", compression: "DEFLATE" });
+
+        if (fs.existsSync(targetPath)) {
+            fs.unlinkSync(targetPath);
+        }
+
+        fs.writeFileSync(targetPath, buffer);
+    }
+
+    static computeSha1FromZip(zipPath: string): string {
+        Guard.argumentNotNullOrEmpty(zipPath, "zipPath");
+
+        if (!fs.existsSync(zipPath)) {
+            throw new Error(`${zipPath} doesn't exist.`);
+        }
+
+        let buffer = fs.readFileSync(zipPath);
+        return sha1(buffer);
+    }
+
+    static getVersionFromReleaseNote(releaseNotePath: string): string {
+        Guard.argumentNotNullOrEmpty(releaseNotePath, "releaseNotePath");
+
+        if (!fs.existsSync(releaseNotePath)) {
+            throw new Error(`${releaseNotePath} doesn't exist.`);
+        }
+
+        let regex = /\(Current\s+Version:\s+[vV]([\d\.]+)\)/;
+        let content = fs.readFileSync(releaseNotePath, "utf8");
+
+        let match = regex.exec(content);
+        if (!match || match.length < 2) {
+            throw new Error(`Can't parse version from ${releaseNotePath} in current version part.`);
+        }
+
+        return match[1].trim();
+    }
+
+    static getDescriptionFromReleaseNote(releaseNotePath: string): string {
+        Guard.argumentNotNullOrEmpty(releaseNotePath, "releaseNotePath");
+
+        if (!fs.existsSync(releaseNotePath)) {
+            throw new Error(`${releaseNotePath} doesn't exist.`);
+        }
+
+        let regex = /---{3,}\r?\n([\s\S]+?)[vV][\d\.]+\r?\n---{3,}\r?\n/;
+        let content = fs.readFileSync(releaseNotePath, "utf8");
+
+        let match = regex.exec(content);
+        if (!match || match.length < 2) {
+            throw new Error(`Can't parse description from ${releaseNotePath} in current version part.`);
+        }
+
+        return match[1].trim();
+    }
+
+    static async isReleaseNoteUpdatedAsync(gitRootPath: string, releaseNotePath: string): Promise<boolean> {
+        Guard.argumentNotNullOrEmpty(gitRootPath, "gitRootPath");
+        Guard.argumentNotNullOrEmpty(releaseNotePath, "releaseNotePath");
+
+        let files = await this.getLatestUpdatedFilesAsync();
+        if (!files) {
+            return false;
+        }
+
+        for (let file in files) {
+            if (path.join(gitRootPath, file) === releaseNotePath) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    static async getLatestUpdatedFilesAsync(): Promise<Array<string>> {
+        let result = await cp.exec("git diff-tree --no-commit-id --name-only -r HEAD");
+        let content = result.stdout.trim();
+        if (!content) {
+            return null;
+        }
+
+        return content.split(/\r?\n/);
+    }
+}
+
+export class Guard {
+    static argumentNotNull(argumentValue: Object, argumentName: string, message = null) {
+        if (argumentValue === null || argumentValue === undefined) {
```

Comment:
>null [](start = 30, length = 4)

suggest reading [https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined)